### PR TITLE
Louise osborne code review

### PR DESCRIPTION
**Pull request by StuartCowley**

### CODE REVIEW

File: src/kata2.booleanToWord.js [hunk 1]
```diff
@@ -1,3 +1,7 @@
-const booleanToWord = (boolean) => {};
-
+const booleanToWord = (boolean) => {
+  if (boolean) {
+    return "Yes";
+  }
+  return "No";
+};
```
Comment (StuartCowley) [inline src/kata2.booleanToWord.js:6]: You can refactor this a little to make it into a one liner using a ternary operator:
```
const booleanToWord = (boolean) => (boolean === true ? "Yes" : "No");
```

File: src/kata3.numberToReversedDigits.js [hunk 2]
```diff
@@ -1,3 +1,6 @@
-const numberToReversedDigits = (number) => {};
-
+const numberToReversedDigits = (number) => {
+  const reversedArray = Array.from(String(number), n => Number(n));
+  reversedArray.reverse();
+  return reversedArray;
+};
```
Comment (StuartCowley) [inline src/kata3.numberToReversedDigits.js:5]: You can reduce this down a bit by not declaring the variables at all, and just returning the chain of existing logic together. Dos the same job in a bit less space while still being readable:
```
const numberToReversedDigits = (number) =>
  Array.from(String(number), (n) => Number(n)).reverse();
```

File: src/kata7.getEmployerRole.js [hunk 3]
```diff
@@ -1,3 +1,8 @@
-const getEmployerRole = (employeeName, employees) => {};
+const getEmployerRole = (employeeName, employees) => {
+  const anEmployee = employees.find(
+    (employee) => employee.name === employeeName
+  );
+  return anEmployee.role;
+};
```
Comment (StuartCowley) [inline src/kata7.getEmployerRole.js:6]: As above this can be simplified into a readable one liner:
```
const getEmployerRole = (employeeName, employees) =>
  employees.find((employee) => employee.name === employeeName).role;
```

File: test/kata6.joinNames.test.js [hunk 4]
```diff
@@ -1,5 +1,20 @@
 const { joinNames } = require("../src");
 
 describe("joinNames", () => {
-  test("returns string of names, seperated by commas and an ampersand", () => {});
+  test("returns string of names, seperated by commas and an ampersand", () => {
+    expect(
+      joinNames([{ name: "Bart" }, { name: "Lisa" }, { name: "Maggie" }])
+    ).toBe("Bart, Lisa & Maggie");
+  });
+  test("returns string of names, seperated by commas and an ampersand", () => {
```
Comment (StuartCowley) [inline test/kata6.joinNames.test.js:9]: Common convention is to leave a clear line between each of your tests to make them easier to read at a glance (also a typo in 'separated'):
```
  });

  test("returns string of names, separated by commas and an ampersand", () => {
```

File: test/kata4.humanCatDogYears.test.js [hunk 5]
```diff
@@ -1,3 +1,13 @@
...
-// Look Ma, no handlebars!!!
+describe("humanCatDogYears", () => {
+  it("returns an array of three numbers, the human years, the equivalent cat years and dog years", () => {
+    expect(humanCatDogYears(10)).toEqual([10, 56, 64]);
+  });
+
+  it("returns an array of three numbers, the human years, the equivalent cat years and dog years", () => {
+    expect(humanCatDogYears(20)).toEqual([20, 96, 114]);
+  });
+});
```
Comment (StuartCowley) [inline test/kata4.humanCatDogYears.test.js:11]: I would put both these assertions together, as you are testing for the same thing thing in effect (passing in non zero values gives you non zero results). You could add another test to assert you get all zeroes back in the event of passing in zero, which is a valid case thats different enough to this set to be worthy of it (as exposes a but on the code as it happens):
```
describe("humanCatDogYears", () => {
  it("returns an array of three numbers, the human years, the equivalent cat years and dog years", () => {
    expect(humanCatDogYears(10)).toEqual([10, 56, 64]);
    expect(humanCatDogYears(20)).toEqual([20, 96, 114]);
  });

  it("returns an age of 0 for all species when passing in 0", () => {
    expect(humanCatDogYears(0)).toEqual([0, 0, 0]);
  });
});
```

File: src/kata4.humanCatDogYears.js [hunk 6]
```diff
@@ -1,3 +1,25 @@
...
+    if (humanAge <= 1) {
+      return 15;
+    }
+    if (humanAge <= 2) {
+      return 24;
+    }
+    return 24 + (humanAge - 2) * 5;
+  }
+  return [humanYears, catYears(number), dogYears(number)];
+};
```
Comment (StuartCowley) [inline src/kata4.humanCatDogYears.js:23]: So theres a few things to look at here. First off, the logic here will fail if you pass in 0 to the function as it will return `[0,15,15]` when it should return `[0,0,0]`. You can fix this by putting something called an early exit condition at the start of the function:
```
  if (number === 0) return [0, 0, 0];
```
Next, theres no point in assigning the value of `number` to a new `humanYears` variable, you can just use `number` as it is

We should also use the same function syntax throughout, since the main function itself is written in arrow syntax the internal functions should be too. Also, you should check the value against a specific number in the if statements if it only applies to one value, eg:
```
  const catYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 4;
  };
```
Putting all this together, you end up with this:
```
const humanCatDogYears = (number) => {
  if (number === 0) return [0, 0, 0];

  const catYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 4;
  };

  const dogYears = (humanAge) => {
    if (humanAge === 1) {
      return 15;
    }
    if (humanAge === 2) {
      return 24;
    }
    return 24 + (humanAge - 2) * 5;
  };
  return [number, catYears(number), dogYears(number)];
};
```

Just a bit tidier :)

**Review by StuartCowley — COMMENTED** [review]
Good work, all cleanly written and working (except for one bug in the humanCatDog years function, which Ive explained and put a solution for), keep it up!

File: src/kata4.humanCatDogYears.js (hunk 6 above)
Comment (ghost) [reply src/kata4.humanCatDogYears.js:23]: Makes sense, thank you!
